src/ComposerRequireChecker/DependencyGuesser/GuessFromComposerAutoloader.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace ComposerRequireChecker\DependencyGuesser;
+
+
+use Composer\Autoload\ClassLoader;
+
+class GuessFromComposerAutoloader implements GuesserInterface
+{
+
+    /**
+     * @var ClassLoader
+     */
+    private $composerAutoloader;
+
+    private $configVendorDir;
+
+    public function __construct(string $composerJsonPath)
+    {
+        $composerJson = json_decode(file_get_contents($composerJsonPath), true);
+        $this->configVendorDir = $this->normalizePath(dirname($composerJsonPath) . '/' . ($composerJson['config']['vendor-dir'] ?? 'vendor'));
+        $this->composerAutoloader = include $this->configVendorDir . '/autoload.php';
```

Comment:
This is very risky, and it can lead to pre-loading analysed code. So far, the library limited itself to using the AST.

If you need to add composer-based file location, maybe check https://github.com/Roave/BackwardCompatibilityCheck/pull/102, where I implemented an abstraction for `PSR-4`, `PSR-0`, `classmap` and `files` loading via BetterReflection. Still extremely slow, but since we need such an abstraction in BetterReflection itself, putting efforts there would be better.